ui.js

```diff
@@ -125,28 +162,23 @@ async function main(task, test=false) {
   task = await getTask(task, options);
 
   // Get the summaries of the files in the directory
-  const summaries = await getSummaries(dir, test);
-  const chunkedSummaries = chunkSummaries(summaries, maxSummaryTokenCount);
+  const allSummaries = await getSummaries(dir, test);
+  // Separate summaries into chunks
+  const chunkedSummaries = chunkSummaries(allSummaries, maxSummaryTokenCount);
  
-  let relevantFiles=[]
-  for (const summaries of chunkedSummaries){
-    // Decide which files are relevant to the task
-    reply = await runAgent(agents.getFiles,task, summaries, interactive);
-    relevantFiles = relevantFiles.concat(reply.output.relevantFiles)
-  }
-  const files = getFiles(relevantFiles)
-  if (files.length == 0) throw new Error("No relevant files found")
+  // Get relevant files (getFiles agent)
+  const relevantFiles = await getRelevantFiles(task, chunkedSummaries, interactive)
 
-  // Ask an agent about each file
-  let solutions = [];
-  for (const file of files) {
-    const coderRes = await runAgent(agents.coder, task, file, interactive);
-    solutions.push({file:file.path, code:coderRes})
+  // Work on solutions
+  let solutions
 
-    if (autoApply){
-      // This actually applies the solution to the file
-      updateFile(file.path, coderRes);
-    }
+  // Depending on Suggestion Mode call different agents
+  if (!suggestionMode) {
+    // Ask agent for code changes (coder agent)
+    solutions = await getSolution(relevantFiles, autoApply)
```

Comment:
The arguments here don't match the function definition of getSolution :/ 

That's why typescript is so much better